assets/js/search-page.js

```diff
@@ -72,15 +73,56 @@ function loadIndex () {
   }
 }
 
-function saveIndex (index) {
+async function saveIndex (index) {
   try {
-    const serializedIndex = JSON.stringify(index)
+    const serializedIndex = await compress(index)      
     sessionStorage.setItem(indexStorageKey(), serializedIndex)
   } catch (error) {
     console.error('Failed to save index: ', error)
   }
 }
 
+async function compress(index) {
+  
+  const stream = new Blob([JSON.stringify(index)], {
+    type: 'application/json',
+  }).stream().pipeThrough(new CompressionStream('gzip'));
+
+  const blob = await new Response(stream).blob()
+  const buffer = await blob.arrayBuffer()
+  return b64encode(buffer)
+}
+
+
+async function decompress(index) {
+  const stream = new Blob([b64decode(index)], {
+    type: "application/json",
+  }).stream().pipeThrough(new DecompressionStream('gzip'));
+
+  const blob = await new Response(stream).text()
+  return JSON.parse(blob)
+}
+
+function b64encode(buffer) {
+    var binary = '';
+    var bytes = new Uint8Array( buffer );
+    var len = bytes.byteLength;
+    for (var i = 0; i < len; i++) {
+        binary += String.fromCharCode( bytes[ i ] );
+    }
+    return window.btoa( binary );
+}
+
+function b64decode(str) {
+  const binary_string = window.atob(str);
+  const len = binary_string.length;
+  const bytes = new Uint8Array(new ArrayBuffer(len));
+  for (let i = 0; i < len; i++) {
+    bytes[i] = binary_string.charCodeAt(i);
+  }
+  return bytes;
+}
+
 function indexStorageKey () {
   return `index:${getProjectNameAndVersion()}`
```

Comment:
```suggestion
  return `idv2:${getProjectNameAndVersion()}`
```

Let's change the key just to ensure we don't accidentally load a past index through compression.